resilience4j-vavr/src/main/java/io/github/resilience4j/decorators/VavrDecorators.java

```diff
@@ -0,0 +1,215 @@
+/*
+ *
+ *  Copyright 2020: KrnSaurabh
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *
+ */
+package io.github.resilience4j.decorators;
+
+import io.github.resilience4j.bulkhead.Bulkhead;
+import io.github.resilience4j.bulkhead.VavrBulkhead;
+import io.github.resilience4j.cache.Cache;
+import io.github.resilience4j.cache.VavrCache;
+import io.github.resilience4j.circuitbreaker.CircuitBreaker;
+import io.github.resilience4j.circuitbreaker.VavrCircuitBreaker;
+import io.github.resilience4j.core.CheckFunctionUtils;
+import io.github.resilience4j.ratelimiter.RateLimiter;
+import io.github.resilience4j.ratelimiter.VavrRateLimiter;
+import io.github.resilience4j.retry.Retry;
+import io.github.resilience4j.retry.VavrRetry;
+import io.vavr.CheckedFunction0;
+import io.vavr.CheckedFunction1;
+import io.vavr.CheckedFunction2;
+import io.vavr.CheckedRunnable;
+
+import java.util.List;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+public interface VavrDecorators {
```

Comment:
I have added `resilience4j-vavr` module as test dependency wherever needed. Here since we moved `vavr` related decorators in `resilience4j-vavr`, I thought its better to move related tests along for the ease of visibility.